src/main/java/com/iridium/iridiumskyblock/gui/IslandTopGUI.java

```diff
@@ -45,24 +55,43 @@ public IslandTopGUI() {
-        Inventory inventory = Bukkit.createInventory(this, topInventoryConfig.size, StringUtils.color(topInventoryConfig.title));
-
-        addContent(inventory);
-
         return inventory;
     }
 
     @Override
     public void addContent(Inventory inventory) {
-        inventory.clear();
         InventoryUtils.fillInventory(inventory, IridiumSkyblock.getInstance().getInventories().islandTopGUI.background);
 
-
         for (int slot : IridiumSkyblock.getInstance().getConfiguration().islandTopSlots.values()) {
             if (islandSlots.containsKey(slot)) {
                 Island island = islandSlots.get(slot);
-                inventory.setItem(slot, ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getInventories().islandTopGUI.item, new PlaceholderBuilder().applyIslandPlaceholders(island).build()));
+                List<Placeholder> placeholderList = new PlaceholderBuilder().applyIslandPlaceholders(island).build();
+                if (IridiumSkyblock.getInstance().getInventories().islandTopGUI.item.material == XMaterial.PLAYER_HEAD
+                        && IridiumSkyblock.getInstance().getInventories().islandTopGUI.item.headData == null) {
+                    Item initialItem = new Item(
+                            XMaterial.PLAYER_HEAD,
+                            slot,
+                            SkinUtils.getHeadData(island.getOwner().getUuid()),
+                            IridiumSkyblock.getInstance().getInventories().islandTopGUI.item.amount,
+                            IridiumSkyblock.getInstance().getInventories().islandTopGUI.item.displayName,
+                            IridiumSkyblock.getInstance().getInventories().islandTopGUI.item.lore
+                    );
+                    boolean querySkin = false;
+                    if (initialItem.headData == null) {
+                        initialItem.headData = IridiumSkyblock.getInstance().getConfiguration().headDataLoadingSkin;
+                        querySkin = true;
+                    }
+                    inventory.setItem(slot, ItemStackUtils.makeItem(initialItem, placeholderList));
+                    if (querySkin) Bukkit.getScheduler().runTaskAsynchronously(IridiumSkyblock.getInstance(), () -> {
+                        String headData = SkinUtils.queryHeadData(island.getOwner().getUuid());
+                        if (headData == null) return;
+                        initialItem.headData = headData;
+                        ItemStack itemStackWithSkin = ItemStackUtils.makeItem(initialItem, placeholderList);
+                        Bukkit.getScheduler().runTask(IridiumSkyblock.getInstance(), () -> inventory.setItem(slot, itemStackWithSkin));
+                    });
+                } else {
+                    inventory.setItem(slot, ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getInventories().islandTopGUI.item, new PlaceholderBuilder().applyIslandPlaceholders(island).build()));
+                }
```

Comment:
Duplicated code, is there no way to unify this?

Reply:
How about making a pr on [IridiumCore](https://github.com/Iridium-Development/IridiumCore) adding a method getPlayerHead(Player) or smthn? if not ill add that when I get home on wednesday